src/google_cloud_debugger/google_cloud_debugger_test/dbg_breakpoint_test.cc

```diff
@@ -215,19 +261,46 @@ TEST_F(DbgBreakpointTest, TrySetBreakpoint) {
 
   // Push the methods into the method vector that
   // the document index matching this Breakpoint will return.
-  pdb_file_fixture_.first_doc_.methods_.push_back(method);
-  pdb_file_fixture_.first_doc_.methods_.push_back(method2);
+  first_doc_.methods_.push_back(method);
+  first_doc_.methods_.push_back(method2);
+
+  SetUpBreakpoint();
 
   EXPECT_TRUE(breakpoint_.TrySetBreakpoint(&file_mock_));
   EXPECT_EQ(breakpoint_.GetILOffset(), il_offset);
   EXPECT_EQ(breakpoint_.GetMethodDef(), method_def);
 }
 
+// Test the TrySetBreakpoint function of DbgBreakpoint when there are
+// multiple documents.
+TEST_F(DbgBreakpointTest, TrySetBreakpointMultipleFilesOne) {
+  std::vector<std::string> other_file_names;
+  other_file_names.push_back("different_program.cs");
+  other_file_names.push_back("test/aprogram.cs");
+  TestMatchingFileName("program.cs", other_file_names);
+}
+
+// Test the TrySetBreakpoint function of DbgBreakpoint when there are
+// multiple documents.
+TEST_F(DbgBreakpointTest, TrySetBreakpointMultipleFilesTwo) {
+  std::vector<std::string> other_file_names;
+  other_file_names.push_back("blah/program.cs");
+  other_file_names.push_back("program.cs");
+  TestMatchingFileName("test/program.cs", other_file_names);
+}
+
+// Test the TrySetBreakpoint function of DbgBreakpoint when there are
+// multiple documents.
+TEST_F(DbgBreakpointTest, TrySetBreakpointMultipleFilesThree) {
+  std::vector<std::string> other_file_names;
+  other_file_names.push_back("wrong_src/test/program.cs");
+  other_file_names.push_back("blah/test/program.cs");
+  TestMatchingFileName("src/test/program.cs", other_file_names);
+}
```

Comment:
What happens if we get the following?  It might be nice to be explicit here with a test

std::vector<std::string> other_file_names;
  other_file_names.push_back("one/src/test/program.cs");
  other_file_names.push_back("two/src/test/program.cs");
  TestMatchingFileName("src/test/program.cs", other_file_names);

Reply:
It will just select the first one (which is "src/test/program.cs") among "src/test/program.cs", "one/src/test/program.cs" and "two/src/test/program.cs".

Reply:
Actually, the way we are setting up the test, the first document actually does not correspond to the `chosen_file_name` so the test above will fail.